### PR TITLE
ELEC-120 Set up Bazel Build System

### PR DESCRIPTION
**Pull request by ckitagawa**
Sets up the Bazel Build system, updated README and .gitignore.

### CODE REVIEW

**Comment by ckitagawa** [comment]
Karl, is this enough for you to set up the Travis CI integration? I think the only command you need to invoke is 

```bash
bazel test ...
```

File: README.md [hunk 1]
```diff
@@ -1,2 +1,84 @@
 # serialite
-Lightweight Golang Serial/Uart interface for Linux
+Lightweight Golang Serial/UART interface for Linux/OS X written entirely in
+native Go.
+
+## Usage
+
+The folder structure is design such that this package can be fetched using
```
Comment (tituschow) [inline README.md]: s/design/designed/

File: README.md [hunk 2]
```diff
@@ -1,2 +1,84 @@
...
+
+```bash
+go get github.com/uw-midsun/serialite
+```
+
+And will automatically be available as an external library for any project.
```
Comment (tituschow) [inline README.md]: And it will?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,84 @@
 # serialite
-Lightweight Golang Serial/Uart interface for Linux
+Lightweight Golang Serial/UART interface for Linux/OS X written entirely in
```
Comment (tituschow) [inline README.md]: Technically macOS ;)

File: README.md [hunk 4]
```diff
@@ -1,2 +1,84 @@
...
+This project depends on [golang.org/x/sys](https://godoc.org/golang.org/x/sys)
+which is a set of low level OS interface commands for Golang. It is well
+maintained and supports Windows, OS X and Linux on various architectures.
+
+## Future Features
+
+These features are under consideration:
+
+*   Windows Support
+*   Hardware and Software Flow Control
```
Comment (tituschow) [inline README.md]: Probably unnecessary, nothing we use actually uses flow control as of now.

File: README.md (hunk 1 above)
Comment (ckitagawa) [inline README.md]: Done.

File: README.md (hunk 2 above)
Comment (ckitagawa) [inline README.md]: Done.

File: README.md (hunk 3 above)
Comment (ckitagawa) [inline README.md]: Done.

File: README.md (hunk 4 above)
Comment (ckitagawa) [inline README.md]: Fair enough, I'll leave it in here in case I feel bored one day and want to add it.

**Pull request merged**